org-code-javabuilder/theater/src/main/java/org/code/theater/Prompter.java

```diff
@@ -1,28 +1,43 @@
 package org.code.theater;
 
 import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicInteger;
 import org.code.media.Image;
-import org.code.protocol.ClientMessageDetailKeys;
-import org.code.protocol.GlobalProtocol;
-import org.code.protocol.OutputAdapter;
+import org.code.protocol.*;
 
 public class Prompter {
+  private static final String FILE_NAME = "prompterImage-";
+  private static final AtomicInteger FILE_INDEX = new AtomicInteger(0);
```

Comment:
for my information: why do we need `AtomicInteger` over a regular counter? Is it for concurrency purposes?

Reply:
Yeah, just an extra layer of concurrency protection. Practically speaking, this should all be single threaded anyway, but this just guarantees we don't have multiple files with the same name.